airflow/executors/base_executor.py

```diff
@@ -104,7 +104,6 @@ class BaseExecutor(LoggingMixin):
     :param parallelism: how many jobs should run at one time. Set to ``0`` for infinity.
     """
 
-    supports_ad_hoc_ti_run: bool = False
```

Comment:
`BaseExecutor` is a part of the public interface of Airflow, is it safe to remove this config?

We don't use this config in Airflow, but we don't know if the users use it in custom plugins or custom executors.

Reply:
I tend to agree; maybe a deprecation would be better.

Reply:
The Run task api/UI feature was certainly public and we ripped that out without too much worry.
I don't think there will be any impact for removing this relatively obscure field that was not used elsewhere. What we uphold to the public is that this interface, when implemented, will plug into Airflow. Worrying about removing this particular field is falling prey to [XKCD 1172](https://xkcd.com/1172/) IMHO.